features/cells/src/main/java/com/wire/android/feature/cells/ui/CellViewModel.kt

```diff
@@ -276,19 +277,20 @@ class CellViewModel @Inject constructor(
         _menu.emit(menuOption)
     }
 
-    private fun onMenuFileAction(file: CellNodeUi.File, action: BottomSheetAction.File) {
-        when (action.action) {
+    private fun onMenuFileAction(file: CellNodeUi.File, action: BottomSheetAction) {
+        when ((action as BottomSheetAction.File).action) {
```

Comment:
Suggestion: I see we are doing this casts in several places, and the action is always present in the 2 subclasses. Does it make sense to have action as a property of the base class `BottomSheetAction` and no casting would be needed. Of course, the enums `FileAction` and `FolderAction` would need their own hierarchy.